src/BTVNanoCommissioning/workflows/QCD_soft_mu_validation.py

```diff
@@ -0,0 +1,524 @@
+        trig_arrs = [
+            events.HLT[_trig] for _trig in triggers if hasattr(events.HLT, _trig)
+        ]
+        req_trig = np.zeros(len(events), dtype="bool")
+        for t in trig_arrs:
+            req_trig = req_trig | t
+        req_lumi = np.ones(len(events), dtype="bool")
+        if isRealData:
+            req_lumi = self.lumiMask(events.run, events.luminosityBlock)
+        if shift_name is None:
+            output = dump_lumi(events[req_lumi], output)
+        ## Jet cuts
+        events.Jet = events.Jet[jet_cut(events, self._campaign)]
+        req_jets = ak.count(events.Jet.pt, axis=1) >= 1
+
+        event_level = ak.fill_none(req_lumi & req_trig & req_jets, False)
+        if len(events[event_level]) == 0:
+            return {dataset: output}
+
+        dxySigcut = 0
+        muNeEmSum = 1
+
+        soft_muon = events.Muon[
+            softmu_mask(events, self._campaign)
+            & (abs(events.Muon.dxy / events.Muon.dxyErr) > dxySigcut)
+            & (events.Muon.pt > 5)
+        ]
+        req_softmu = ak.count(soft_muon.pt, axis=1) >= 1
+        event_jet = events.Jet
+        mujetsel = ak.fill_none(
+            (
+                (ak.all(event_jet.metric_table(soft_muon) <= 0.4, axis=2))
+                & ((event_jet.muonIdx1 != -1) | (event_jet.muonIdx2 != -1))
+                & ((event_jet.muEF + event_jet.neEmEF) < muNeEmSum)
+                & (event_jet.pt > 20)
+                & ((event_jet.pt / event_jet.E) > 0.03)
+            ),
+            False,
+            axis=-1,
+        )
+        soft_muon = ak.pad_none(soft_muon, 1, axis=1)
+        soft_muon["dxySig"] = soft_muon.dxy / soft_muon.dxyErr
+
+        ## Muon-jet cuts
+        event_jet["isMuonJet"] = mujetsel
+        mu_jet = event_jet[mujetsel]
+        otherjets = event_jet[~mujetsel]
+        req_mujet = ak.num(mu_jet.pt, axis=1) >= 1
+        mu_jet = ak.pad_none(mu_jet, 1, axis=1)
+
+        ####################
+        # Selected objects #
+        ####################
+        event_level = event_level & req_softmu & req_mujet
+        event_level = ak.fill_none(event_level, False)
+        sjets = events.Jet[event_level]
+        ssmu = soft_muon[event_level]
+        smuj = mu_jet[event_level]
+        nssmu = ak.count(ssmu.pt, axis=1)
+        nsmuj = ak.count(smuj.pt, axis=1)
+        njet = ak.count(sjets.pt, axis=1)
+
+        osss = 1
+        ###############
+        # Selected SV #
+        ###############
+        selev = events[event_level]
+        if len(events[event_level]) == 0:
+            if self.isArray:
+                array_writer(
+                    self,
+                    events[event_level],
+                    events,
+                    "nominal",
+                    dataset,
+                    isRealData,
+                    empty=True,
+                )
+            return {dataset: output}
+        ### FIXME: Check if the JetSVs are present in the events

```

Comment:
can you please also link to the git issues

Reply:
Added a FIXME and a link